components/provisioner/internal/model/gardener_config.go

```diff
@@ -612,6 +648,15 @@ func (c AWSGardenerConfig) ExtendShootConfig(gardenerConfig GardenerConfig, shoo
 		return apperrors.Internal("error encoding control plane config: %s", err.Error())
 	}
 
+	if c.input.EnableIMDSv2 != nil && *c.input.EnableIMDSv2 {
+		awsWorkerConfig := NewAWSWorkerConfig(2)
```

Comment:
Same here, passing the magic number to this function could led to the confusion. We could think about a good way of passing this value, but it up to you.

Reply:
Introduced global variable for AWS IMDSv2 HTTPPutResponseHopLimit, and it's used everywhere instead of the "2" magic number.